src/librustdoc/html/static/search.js

```diff
@@ -975,26 +975,32 @@ window.initSearch = function(rawSearchIndex) {
             output = "<div class=\"search-results " + extraClass + "\">";
 
             array.forEach(function(item) {
-                var name, type;
-
-                name = item.name;
-                type = itemTypes[item.ty];
-
                 if (item.is_alias !== true) {
                     if (duplicates[item.fullPath]) {
                         return;
                     }
                     duplicates[item.fullPath] = true;
                 }
+
+                var name = item.name;
+                var type = itemTypes[item.ty];
+
                 length += 1;
 
+                var extra = "";
+                if (type === "primitive") {
+                    extra = " <i>(primitive type)</i>";
+                } else if (type === "keyword") {
+                    extra = " <i>(keyword)</i>";
+                }
```

Comment:
Why declare a separate variable rather than add these directly to `name`?

Reply:
Simpler to read in my opinion. So as your prefer. :)

Reply:
If you prefer this way I'm fine with it. :-) Will r+ once conflicts are resolved.

Reply:
Ok haha.